src/wasm/wasm-validator.cpp

```diff
@@ -2799,17 +2799,19 @@ static void validateMemory(Module& module, ValidationInfo& info) {
           continue;
         }
       }
-      info.shouldBeTrue(checkSegmentOffset(segment.offset,
-                                           segment.data.size(),
-                                           curr.initial * Memory::kPageSize),
-                        segment.offset,
-                        "memory segment offset should be reasonable");
-      if (segment.offset->is<Const>()) {
-        auto start = segment.offset->cast<Const>()->value.getUnsigned();
-        auto end = start + size;
-        info.shouldBeTrue(end <= curr.initial * Memory::kPageSize,
-                          segment.data.size(),
-                          "segment size should fit in memory (end)");
+      if (!module.memory.imported()) {
```

Comment:
this validation change is not obvious to me. why, and how is it related to module linking in wasm-shell?

Reply:
`data.wast` spec test fails without this for imported memory test cases. Same goes for imported tables. Looks like they should be checked at runtime, where the actual size of the memory or table instance is known.

In any case, I reverted all spec test changes from this PR to keep it focused on module linking.